R/preproc.R

```diff
@@ -1,19 +1,32 @@
+library(tidyverse)
+library(dplyr)
+library(fastDummies)
 #' Preproc
 #'
-#' @param X_train (dpylyr dataframe) X_train dateframe - Required
-#' @param X_test (dpylyr dataframe)  X_test dataframe - Default None
+#' @param X (dpylyr dataframe) X_train dateframe - Required
 #'
-#' @return (tuple) Tuple containing two dataframes (X_train, X_test)
+#' @return (dataframe) Preprocessed df with scaling or encoding
 #'
 #' @export
 #'
 #' @examples
-#' X_train <- vector(mode="list", length=1)
-#' X_test <- vector(mode="list", length=1)
-#' result = preproc(X_train, X_test)
-#' processed_X_train = result[0]
-#' processed_X_test = result[1]
-  preproc <- function(X_train, X_test=NULL) {
-  X_train
+#' result = preproc(mtcars)
+#' processed_X = result
+  preproc <- function(X) {
+    if (!is.data.frame(X) & !is_tibble(X)) {
```

Comment:
is.data.frame and is_tibble do not always capture the correct class of X - this may require further investigation

Reply:
I think this covers tibbles or dfs, and should be safe. Im going to resolve this for now. If it apepars to be a problem still we can discuss in the next iteration of milestone 3, cheers.